types/index.d.ts

```diff
@@ -3,11 +3,11 @@
 // Definitions by: Rahim Alwer <https://github.com/mihar-22>
 
 import {queries, Queries, BoundFunction, EventType} from '@testing-library/dom'
-import { SvelteComponent, ComponentProps } from 'svelte/types/runtime'
+import { SvelteComponent, ComponentProps, ComponentConstructorOptions } from 'svelte/types/runtime'
 
 export * from '@testing-library/dom'
 
-type SvelteComponentOptions<C extends SvelteComponent> = ComponentProps<C> | {props: ComponentProps<C>}
+type SvelteComponentOptions<C extends SvelteComponent> = ComponentProps<C> | Pick<ComponentConstructorOptions<ComponentProps<C>>, "anchor" | "props" | "hydrate" | "intro" | "context">
```

Comment:
Looking at [this line](https://github.com/testing-library/svelte-testing-library/blob/main/src/pure.js#L12) and [this unit test](https://github.com/testing-library/svelte-testing-library/blob/main/src/__tests__/render.test.js#L43), I'd say it definetely should be there because it's a valid option; however not as a `Pick` I think, because it [does not exist on ComponentConstructorOptions](https://github.com/sveltejs/svelte/blob/master/packages/svelte/src/runtime/internal/public.d.ts#L4).

Maybe something like

```
type SvelteComponentOptions<C extends SvelteComponent> = ComponentProps<C> | Pick<ComponentConstructorOptions<ComponentProps<C>>, "anchor" | "props" | "hydrate" | "intro" | "context"> & { accessors?: boolean }
```

?

Reply:
Hey! 

It should not. The `accessors` is not part of the API and does nothing. The test mentioned above works only because it use the `<svelte:options accessors />` as [L1](https://github.com/testing-library/svelte-testing-library/blob/main/src/__tests__/fixtures/Comp.svelte#L1). See official doc about `<svelte:options>` [here](https://svelte.dev/docs/special-elements#svelte-options)) and additionally to `svelte/compiler` options [here](https://svelte.dev/docs/svelte-compiler#types-compileoptions).

The `accessors` config has no impact here. You can set it to `false` in your test case it should still pass the assertions. As mentioned in the doc, Svelte component accessors are built at compile time and not at runtime. For example with a component defined like this:

```svelte
<script>
  export let testId;
</script>
<div data-testid={testId} />
```

The svelte compiler generate this:

```js
// ...generated code for fragment and instance management

// generated class
class Component extends SvelteComponent {
	constructor(options) {
		super();
		init(this, options, instance, create_fragment, safe_not_equal, { testId: 0 });
	}
}

export default Component;
```

Now, if you add `<svelte:options accessors />` to the component, the compiler generates this:

```js
// ...generated code for fragment and instance management

// generated class
class Component extends SvelteComponent {
	constructor(options) {
		super();
		init(this, options, instance, create_fragment, safe_not_equal, { testId: 0 });
	}

	get testId() {
		return this.$$.ctx[0];
	}

	set testId(testId) {
		this.$$set({ testId });
		flush();
	}
}

export default Component;
```

Hope it helps!

Reply:
Additional remark I would not use `Pick` to exclude any props from `ComponentConstructorOptions<T>` since Svelte 4 now exposes only what they consider to be public.